syntax = "proto3";
package alert;




import "monitor_sdk/model/monitor/alert_event.proto";

/*
GetAlertEventList请求
*/
message GetAlertEventListRequest {
    /*
    告警 ID
    */
    string _id = 1 ;
    /*
    选择返回的字段，以逗号分隔
    */
    string __select__ = 2 ;
    /*
    页码
    */
    int32 page = 3 ;
    /*
    分页大小
    */
    int32 page_size = 4 ;
    /*
    模型id
    */
    string objectId = 5 ;
    /*
    实例id
    */
    string instanceId = 6 ;
    /*
    告警策略ID
    */
    string rule_id = 7 ;
    /*
    开始时间, e.g.: -1h 过去1小时, -1d 过去1天, -30m 过去30分钟, 1490967693 unix 时间戳
    */
    string st = 8 ;
    /*
    结束时间, e.g.: -1h 过去1小时, -1d 过去1天, -30m 过去30分钟, 1490967693 unix 时间戳。默认为当前时间
    */
    string et = 9 ;
}

/*
GetAlertEventList返回
*/
message GetAlertEventListResponse {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    消息
    */
    string msg = 2 ;
    /*
    页码
    */
    int32 page = 3 ;
    /*
    分页大小
    */
    int32 page_size = 4 ;
    /*
    总数
    */
    int32 total = 5 ;
    /*
    告警事件列表
    */
    repeated monitor.AlertEvent data = 6 ;
}

/*
GetAlertEventListApi返回
*/
message GetAlertEventListResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    alert.GetAlertEventListResponse data = 4 ;
}

